src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs

```diff
@@ -111,6 +115,16 @@ impl Container {
             None => true,
         };
         let annotations = spec.annotations().clone().unwrap_or_default();
+        let container_typ = container_type(&spec);
+        let annotations_to_add = if container_typ.is_pod_container() {
```

Comment:
Do you envision more annotations added and removed here in future PRs? If it's just one or two, why not just do it directly?